sdcm/utils/data_validator.py

```diff
@@ -493,90 +684,77 @@ def validate_range_expected_to_change(self, session, during_nemesis=False):
         if not during_nemesis:
             LOGGER.debug('Verify updated rows')
 
-        partition_keys = ', '.join(self.base_table_partition_keys)
-
-        for views_set in self.list_of_view_names_for_update_test():
-            # views_set[0] - view name with rows before update
-            # views_set[1] - view name with rows after update
-            # views_set[2] - view name with all expected partition keys
-            # views_set[3] - do perform validation for the view or not
+        # List of tuples of correlated  view names for validation: before update, after update, expected data
+        views_list = list(zip(self.view_names_for_updated_data,
```

Comment:
why not use `list_of_view_names_for_update_test` ?
also why no create a list of `DataForValidation` to begin with ?

Reply:
> looking at the failed run, seem like the critical events doesn't have any information:
> 
> ```
> 2022-09-14 15:06:45.048: (DataValidatorEvent Severity.CRITICAL) period_type=one-time event_id=eb71d16e-ed23-4771-b2a7-91a5a414f32f: type=UpdatedRowsValidator error=
> 2022-09-14 15:06:45.438: (DataValidatorEvent Severity.CRITICAL) period_type=one-time event_id=7639ee67-504e-48ae-a250-bc5ad0b2d2fe: type=UpdatedRowsValidator error=
> ```
> 
> also what exact one should do with the json files that were saved ? how can one understand what exactly is the validation issue ? can you elaborate a bit on that somewhere in /doc/ ?

I fixed it already, just did not re-run the test

Reply:
> why not use `list_of_view_names_for_update_test` ? 
yep, i used this function but it was missed during rebase. fixed

> also why no create a list of `DataForValidation` to begin with ?
It needs less code when create fully filled NameTuple, as named tuples are immutable